### ISSUE TITLE
Invalid "Use of moved value" on Copy type with explicit Copy bound

### ISSUE THREAD

**Issue by afetisov** [intellij-rust/intellij-rust#6953]
## Environment

* **IntelliJ Rust plugin version:** 0.3.142.3705-203
* **Rust toolchain version:** 1.52.0-nightly (45b3c2851 2021-03-04) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2020.3.2 (CL-203.7148.70)
* **Operating system:** Linux 5.4.0-66-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

IDE incorrectly marks copies of a Copy value as "Use of moved value". This only happens when the type is explicitly demanded to be Copy in the function trait bounds.

## Steps to reproduce

```rust
#[derive(Clone, Copy)]
struct Foo;

fn f(x: Foo) where Foo: Copy {
    f(x); // 1
    f(x); // 2
}
```

The IDE borrow checker inspection produces error on line 2: "Use of moved value".

Removing either the derive or the explicit 'where' bound removes the error.

**Comment by msrd0** [comment]
This also happens without the explicit trait bound with primitive types:

![image](https://user-images.githubusercontent.com/7853372/160863466-55caac0e-5c99-4937-8539-011dfd04c4c5.png)

**Comment by afetisov** [comment]
Duplicate of #8439 

Also, seems to be fixed.
